tools/makecorever.py

```diff
@@ -20,23 +20,40 @@
 import argparse
 import os
 import subprocess
+import re
 
 
-def generate(path, platform_path, git_ver="ffffffff", git_desc="unspecified"):
+def generate(path, platform_path, git_ver="ffffffff", platform_version="unspecified"):
     def git(*args):
         cmd = ["git", "-C", platform_path]
         cmd.extend(args)
         proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, universal_newlines=True, stderr=subprocess.DEVNULL)
         return proc.stdout.readlines()[0].strip()
 
+    git_desc = platform_version;
     try:
         git_ver = git("rev-parse", "--short=8", "HEAD")
         git_desc = git("describe", "--tags")
     except Exception:
         pass
 
-    text = "#define ARDUINO_ESP8266_GIT_VER 0x{}\n".format(git_ver)
-    text += "#define ARDUINO_ESP8266_GIT_DESC {}\n".format(git_desc)
+    text = "#define ARDUINO_ESP8266_GIT_VER   0x{}\n".format(git_ver)
+    text += "#define ARDUINO_ESP8266_GIT_DESC  {}\n".format(git_desc)
+    text += "\n"
+
+    version = re.split("\.", platform_version)
+    # major: if present, skip "unix-" in "unix-3"
+    text += "#define ARDUINO_ESP8266_MAJOR     {}\n".format(re.split("-", version[0])[-1])
+    text += "#define ARDUINO_ESP8266_MINOR     {}\n".format(version[1])
+    # revision can be ".n" or ".n-dev"
+    revision = re.split("-", version[2])
+    text += "#define ARDUINO_ESP8266_REVISION  {}\n".format(revision[0])
+    text += "\n"
+    if len(revision) > 1:
+        text += "#define ARDUINO_ESP8266_DEV       1 // developpment version\n"
+    else:
+        text += "#define ARDUINO_ESP8266_RELEASE   \"{}\"\n".format(git_desc)
+        text += "#define ARDUINO_ESP8266_RELEASE_{}\n".format(re.sub("[-\.]", "_", git_desc))
```

Comment:
also, re. `re`
python has string methods like `"1.2.3".split(".")` and `"1.2.3".replace(".", "_")`
also note of the `"1.2.3-dev".partition("-")`